flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/batch/table/ColumnsOperationTest.scala

```diff
@@ -0,0 +1,282 @@
+    val func0 = new TableFunc0
+    util.tableEnv.registerFunction("func0", func0)
+
+    val tab1 = t.joinLateral(func0(columns('string)))
+    val tab2 = t.joinLateral("func0(columns(string))")
+
+    val expected =
+      unaryNode(
+        "DataSetCorrelate",
+        batchTableNode(0),
+        term("invocation",
+          "org$apache$flink$table$utils$TableFunc0$497a630d2a145bca99673bcd05a53d2b($2)"),
+        term("correlate", "table(TableFunc0(string))"),
+        term("select", "int", "long", "string", "name", "age"),
+        term("rowType",
+          "RecordType(DOUBLE int, BIGINT long, VARCHAR(65536) string, VARCHAR(65536) name," +
+            " INTEGER age)"),
+        term("joinType", "INNER")
+      )
+
+    util.verifyTable(tab1, expected)
+    util.verifyTable(tab2, expected)
+  }
+
+  @Test
+  def testFilter(): Unit = {
+    val t = util.addTable[(Int, Long, String, String)]('int, 'long, 'string1, 'string2)
+
+    val tab1 = t.where(concat(columns('string1 ~ 'string2)) === "a")
+    val tab2 = t.where("concat(columns(string1 ~ string2)) = 'a'")
+
+    val expected =
+      unaryNode(
+        "DataSetCalc",
+        batchTableNode(0),
+        term("select", "int", "long", "string1", "string2"),
+        term("where", "=(CONCAT(string1, string2), 'a')")
+      )
+
+    verifyAll(tab1, tab2, expected)
+  }
+
+  @Test
+  def testGroupBy(): Unit = {
+    val t = util.addTable[(Int, Long, String, Int, Long, String)]('a, 'b, 'c, 'd, 'e, 'f)
+
+    val tab1 = t
+      .groupBy(columns(1), 'b)
+      .select('a, 'b, 'c.count)
+
+    val tab2 = t
+      .groupBy("columns(1), b")
+      .select("a, b, c.count")
+
+    val expected =
+      unaryNode(
+        "DataSetAggregate",
+        unaryNode(
+          "DataSetCalc",
+          batchTableNode(0),
+          term("select", "a", "b", "c")
+        ),
+        term("groupBy", "a", "b"),
+        term("select", "a", "b", "COUNT(c) AS TMP_0")
+      )
+
+
+    verifyAll(tab1, tab2, expected)
+  }
+
+
+  @Test
+  def testWindowGroupBy(): Unit = {
+    val t = util.addTable[(Int, Long, String, Int)]('a, 'b, 'c, 'd)
+
+    val tab1 = t
+      .window(Slide over 3.milli every 10.milli on columns('b) as 'w)
+      .groupBy(columns('a, 'b), 'w)
+      .select(columns(1 ~ 2), columns('c).count)
+
+    val tab2 = t
+      .window(Slide.over("3.milli").every("10.milli").on("columns(b)").as("w"))
+      .groupBy("columns(a, b), w")
+      .select("columns(1 ~ 2), columns(c).count")
+
+    val expected =
+      unaryNode(
+        "DataSetWindowAggregate",
+        unaryNode(
+          "DataSetCalc",
+          batchTableNode(0),
+          term("select", "a", "b", "c")
+        ),
+        term("groupBy", "a", "b"),
+        term("window", "SlidingGroupWindow('w, 'b, 3.millis, 10.millis)"),
+        term("select", "a", "b", "COUNT(c) AS TMP_0")
+      )
+    verifyAll(tab1, tab2, expected)
+  }
+
+  @Test
+  def testOrderBy(): Unit = {
```

Comment:
We can only keep OrderBy test, due to other operator can be test by stream/table/ColumnsOperationsTest, What do you think?